torch/testing/_internal/common_utils.py

```diff
@@ -834,30 +888,23 @@ def assertTensorsEqual(a, b):
                             a = a.to(torch.int)
                             b = b.to(torch.int)
 
-                        diff = a - b
-                        if a.dtype.is_complex or a.dtype.is_floating_point:
-                            # check that NaNs are in the same locations
-                            nan_mask = torch.isnan(a)
-                            self.assertTrue(torch.equal(nan_mask, torch.isnan(b)), message)
-                            diff[nan_mask] = 0
-                            # inf check if allow_inf=True
-                            if allow_inf:
-                                inf_mask = torch.isinf(a)
-                                inf_sign = inf_mask.sign()
-                                self.assertTrue(torch.equal(inf_sign, torch.isinf(b).sign()), message)
-                                diff[inf_mask] = 0
-                        # TODO: implement abs on CharTensor (int8)
-                        # TODO: modify abs to return float/double for ComplexFloat/ComplexDouble
-                        if diff.is_signed() and diff.dtype != torch.int8:
-                            diff = diff.abs()
-                            # if diff is complex, the imaginary component for diff will be 0
-                            # from the previous step, hence converting it to float and double is fine.
-                            if diff.dtype == torch.complex64:
-                                diff = diff.to(torch.float)
-                            elif diff.dtype == torch.complex128:
-                                diff = diff.to(torch.double)
-                        max_err = diff.max()
-                        self.assertLessEqual(max_err, prec, message)
+                        if a.is_complex():
+                            # todo: assert_allclose should handle complex types directly.
```

Comment:
as discussed before, we shouldn't compare angles and absolute values, reason being that there might be a substantial difference in the values for same (but very large) absolute value and small angle difference.

I have added the temporary real/imag functions for complex in this PR https://github.com/pytorch/pytorch/pull/35879 so we should instead use that [method calls: ```copy_real(), copy_imag()```]